lib/reservation.rb

```diff
@@ -0,0 +1,43 @@
+require "date"
+require "pry"
+
+class Reservation
+  attr_accessor :reservation_id, :check_in, :check_out, :room, :is_hotel_blocker
+
+  def initialize(reservation_id, check_in: nil, check_out: nil, room: nil, is_hotel_blocker: false)
+    @reservation_id = reservation_id
+    check_in == nil ? @check_in = nil : @check_in = Date.parse(check_in)
```

Comment:
It's easy for someone reading your code to overlook assignment inside of ternaries.

Consider:
```suggestion
    @check_in = check_in == nil ? nil : Date.parse(check_in)
```